specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2024-03-01/amlfilesystem.json

```diff
@@ -0,0 +1,1453 @@
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "AML file system has started the cancellation of the archive operation. Poll the AML file system's hsm.archiveStatus field to monitor the progress."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "storagecache.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "amlFilesystems_cancelArchive": {
+            "$ref": "./examples/amlFilesystems_CancelArchive.json"
+          }
+        },
+        "tags": [
+          "amlFilesystems"
+        ],
+        "description": "Cancel archiving data from the AML file system.",
+        "operationId": "amlFilesystems_CancelArchive"
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.StorageCache/amlFilesystems/{amlFilesystemName}/import": {
+      "post": {
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/AmlFilesystemNameParameter"
+          },
+          {
+            "name": "ImportInfo",
+            "in": "body",
+            "description": "Information about the import operation",
+            "schema": {
+              "$ref": "#/definitions/AmlFilesystemImportInfo"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "AML file system has started the import operation. Poll the AML file system's hsm.importStatus field to monitor the progress."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "storagecache.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "amlFilesystems_Import": {
+            "$ref": "./examples/amlFilesystems_Import.json"
+          }
+        },
+        "tags": [
+          "amlFilesystems"
+        ],
+        "description": "Import data from the AML file system. This operation cannot start when an archive operation is in progress.",
+        "operationId": "amlFilesystems_Import"
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.StorageCache/amlFilesystems/{amlFilesystemName}/cancelImport": {
```

Comment:
>cancelImport [](http://example.com/codeflow?start=140&length=12)

ARM much recommends modelling operations using CRUD : PUT\GET\DELETE. since most tools written on top of ARM only work with these for example ARM templates \ ARM policy \ ARG etc 


Consider if this could be better modelled as a PUT on an ImportJob, which is a long running operation and gives you the status at the end

Reply:
@raosuhas Brian explained above why we chose POST but to expand, the import job will run to completion and then get removed without further interaction needed from the customer. We see an import/cancelImport as an attribute for the AMLFS resource. If this were a sub-resource, we could no longer expose the status/details in a single GET AMLFS, which would not be optimal for our customers.

The import/cancelImport are actions on the amlfs resource that import data to the amlfs cluster. These actions are similar to actions on other resources like ["e.g. getting the thumbnail of an image or rebooting a VM"](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#performing-an-action)